In [1]:
import plotly_express as px
import pandas as pd

# Importing dataset from excel file

In [14]:
df_vaccine_data = pd.read_excel("./Data/folkhalsomyndigheten_Covid19_Vaccine.xlsx", sheet_name= "Vaccinerade kommun och ålder")

In [15]:
df_vaccine_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2900 entries, 0 to 2899
Data columns (total 14 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Län                  2900 non-null   int64  
 1   Län_namn             2900 non-null   object 
 2   Kommun               2900 non-null   int64  
 3   Kommun_namn          2900 non-null   object 
 4   Ålder                2900 non-null   object 
 5   Befolkning           2900 non-null   int64  
 6   Antal minst 1 dos    2900 non-null   int64  
 7   Antal minst 2 doser  2900 non-null   int64  
 8   Antal 3 doser        2320 non-null   float64
 9   Antal 4 doser        870 non-null    float64
 10  Andel minst 1 dos    2900 non-null   float64
 11  Andel minst 2 doser  2900 non-null   float64
 12  Andel 3 doser        2320 non-null   float64
 13  Andel 4 doser        870 non-null    float64
dtypes: float64(6), int64(5), object(3)
memory usage: 317.3+ KB


In [16]:
df_vaccine_data.head(7)

Län        Län_namn  Kommun     Kommun_namn  Ålder  Befolkning  \
0    1  Stockholms län     114  Upplands Väsby  12-15        2422   
1    1  Stockholms län     114  Upplands Väsby  16-17        1203   
2    1  Stockholms län     114  Upplands Väsby  18-29        6692   
3    1  Stockholms län     114  Upplands Väsby  30-39        7332   
4    1  Stockholms län     114  Upplands Väsby  40-49        6946   
5    1  Stockholms län     114  Upplands Väsby  50-59        6332   
6    1  Stockholms län     114  Upplands Väsby  60-69        4258   

   Antal minst 1 dos  Antal minst 2 doser  Antal 3 doser  Antal 4 doser  \
0               1206                 1046            NaN            NaN   
1                839                  755            NaN            NaN   
2               4887                 4469         1959.0            NaN   
3               5542                 5240         2878.0            NaN   
4               5592                 5429         3719.0            NaN   
5               5644                 5558         4535.0            NaN   
6               3829                 3786         3335.0            NaN   

   Andel minst 1 dos  Andel minst 2 doser  Andel 3 doser  Andel 4 doser  
0           0.497936             0.431874            NaN            NaN  
1           0.697423             0.627598            NaN            NaN  
2           0.730275             0.667812       0.292738            NaN  
3           0.755865             0.714675       0.392526            NaN  
4           0.805068             0.781601       0.535416            NaN  
5           0.891346             0.877764       0.716203            NaN  
6           0.899248             0.889150       0.783232            NaN

### Task A
Counties in the dataset

In [17]:
df_vaccine_data["Län_namn"].unique()

array(['Stockholms län', 'Uppsala län', 'Södermanlands län',
       'Östergötlands län', 'Jönköpings län', 'Kronobergs län',
       'Kalmar län', 'Gotlands län', 'Blekinge län', 'Skåne län',
       'Hallands län', 'Västra Götalands län', 'Värmlands län',
       'Örebro län', 'Västmanlands län', 'Dalarnas län', 'Gävleborgs län',
       'Västernorrlands län', 'Jämtlands län', 'Västerbottens län',
       'Norrbottens län'], dtype=object)

In [18]:
df_vaccine_data["Län_namn"].nunique()

21

There are 21 counties in sweden, so the dataset seems to have data about all of them.

### Task B
Municipalities in the dataset

In [22]:
df_vaccine_data["Kommun_namn"].unique()

array(['Upplands Väsby', 'Vallentuna', 'Österåker', 'Värmdö', 'Järfälla',
       'Ekerö', 'Huddinge', 'Botkyrka', 'Salem', 'Haninge', 'Tyresö',
       'Upplands-Bro', 'Nykvarn', 'Täby', 'Danderyd', 'Sollentuna',
       'Stockholm', 'Södertälje', 'Nacka', 'Sundbyberg', 'Solna',
       'Lidingö', 'Vaxholm', 'Norrtälje', 'Sigtuna', 'Nynäshamn', 'Håbo',
       'Älvkarleby', 'Knivsta', 'Heby', 'Tierp', 'Uppsala', 'Enköping',
       'Östhammar', 'Vingåker', 'Gnesta', 'Nyköping', 'Oxelösund', 'Flen',
       'Katrineholm', 'Eskilstuna', 'Strängnäs', 'Trosa', 'Ödeshög',
       'Ydre', 'Kinda', 'Boxholm', 'Åtvidaberg', 'Finspång',
       'Valdemarsvik', 'Linköping', 'Norrköping', 'Söderköping', 'Motala',
       'Vadstena', 'Mjölby', 'Aneby', 'Gnosjö', 'Mullsjö', 'Habo',
       'Gislaved', 'Vaggeryd', 'Jönköping', 'Nässjö', 'Värnamo', 'Sävsjö',
       'Vetlanda', 'Eksjö', 'Tranås', 'Uppvidinge', 'Lessebo', 'Tingsryd',
       'Alvesta', 'Älmhult', 'Markaryd', 'Växjö', 'Ljungby', 'Högsby',
       '

In [23]:
df_vaccine_data["Kommun_namn"].nunique()

290

There are 290 counties in sweden, so the dataset seems to have data about all of them as well.

### Task C
Population in the dataset

In [33]:
vaccine_data_population = df_vaccine_data["Befolkning"].sum()

In [34]:
vaccine_data_population

9092790

It seems that the population in the dataset is 9 092 790 people.

### Task D
Children under age 18

In [30]:
df_vaccine_data_ages = df_vaccine_data["Ålder"].unique()

In [31]:
df_vaccine_data_ages

array(['12-15', '16-17', '18-29', '30-39', '40-49', '50-59', '60-69',
       '70-79', '80-89', '90 eller äldre'], dtype=object)

Looking at the dataset, it doesn't seem to have data about children under the age of 12.
We'll have to use info about whole population of sweden in 2022 to help us find those.
According to SCB the population of sweden was 10 521 556 in 2022.

In [32]:
sweden_population_2022 = 10521556

In [36]:
excluded_age_groups = ["12-15", "16-17"]
adult_population = df_vaccine_data[~df_vaccine_data["Ålder"].isin(excluded_age_groups)]["Befolkning"].sum()

In [37]:
adult_population

8347420

In [38]:
child_population = sweden_population_2022 - adult_population

In [39]:
child_population

2174136

Using the data from SCB we can find out that the population of children under the age of 18 is 2 174 136.

### Task E
Age distribution diagram.

We'll need to add the population of ages 0-11 to the dataset before we can make the diagram.

In [118]:
ages_0_11 = {"Ålder" : "0-11", "Befolkning" : sweden_population_2022 - vaccine_data_population}

In [119]:
df_ages_0_11 = pd.DataFrame([ages_0_11])

In [120]:
df_ages_0_11

Ålder  Befolkning
0  0-11     1428766

In [121]:
df_age_ranges_population = pd.DataFrame({"Ålder" : df_vaccine_data_ages, "Befolkning" : 0})

In [122]:
df_age_ranges_population

Ålder  Befolkning
0           12-15           0
1           16-17           0
2           18-29           0
3           30-39           0
4           40-49           0
5           50-59           0
6           60-69           0
7           70-79           0
8           80-89           0
9  90 eller äldre           0

In [123]:
df_age_ranges_population["Befolkning"] = df_vaccine_data.groupby("Ålder")["Befolkning"].sum().values

In [124]:
df_age_ranges_population

Ålder  Befolkning
0           12-15      503831
1           16-17      241539
2           18-29     1475950
3           30-39     1467590
4           40-49     1298156
5           50-59     1339798
6           60-69     1121922
7           70-79     1033113
8           80-89      496750
9  90 eller äldre      114141

In [127]:
df_age_ranges_population = pd.concat([df_ages_0_11, df_age_ranges_population], ignore_index=True)

In [128]:
df_age_ranges_population

Ålder  Befolkning
0             0-11     1428766
1            12-15      503831
2            16-17      241539
3            18-29     1475950
4            30-39     1467590
5            40-49     1298156
6            50-59     1339798
7            60-69     1121922
8            70-79     1033113
9            80-89      496750
10  90 eller äldre      114141

Creating a pie-chart to show the population by age group.

In [134]:
category_order = df_age_ranges_population["Ålder"].tolist()

age_group_population_chart = px.pie(
    df_age_ranges_population, names="Ålder", values="Befolkning", 
    title="Swedish population by age groups", category_orders={"Ålder": category_order}
)
age_group_population_chart.show()
age_group_population_chart.write_html("./Visualisations/Task_2_plotly_swedish_population_by_age_group.html")